subsys/net/ip/net_tls.h

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
> subsys/net/ip/net_tls.h

Even the name of this file looks weird. There's no TLS in IP. Layering violation.

Reply:
This entire patch series is a layering violation. But, TLS itself is pretty much designed as a layering violation. The best place to describe TLS is as a presentation layer, which sits _entirely_ above the transport layer, and below the application layer.

The only argument I can see for not creating a TLS layer that just calls the socket code is because of the desire to offload work. But, I would argue that we should be offloading at the presentation layer, and not the transport layer. That would mean our API should be there, and, well, something like what net_app already does.